### PR TITLE
Update stack

### PR DESCRIPTION
**Pull request by Priceless-P**
#### Task reference
This section contains a link to a corresponding task in **Linear**.

[link to issue on linear](https://linear.app/team-axle-skript-project/issue/BAC-142/build-endpoint-for-edit-stack)
#### Summary
This section contains a brief description of **_what was accomplished_** within the current PR.

- Made sure a stack can only be updated by the person that created it
- Add middleware to update stack route
- Additional checks
- 
#### Implementation details
This section describes the **_technical aspects_** of the solution, implemented within the current PR, that are worth mentioning to the reviewers.
#### How to test
This section contains a checklist, describing basic scenarios that would help verify that the submitted code works correctly.
#### References 
Any additional information (screenshots, video, diagrams) that would help understand and review the PR.

When everything goes right

Screen Shot 2022-12-02 at 3 12 11 PM

when a user tries to update a stack he didn't recreate

Screen Shot 2022-12-02 at 5 33 43 PM

### CODE REVIEW

File: backend/app/Http/Controllers/StackController.php [hunk 1]
```diff
@@ -12,43 +12,72 @@ class StackController extends Controller
...
-                    'Stack not found',
-                    Response::HTTP_NOT_FOUND
-                );
+            // Get current user
+            $uid = $request->user["id"];
+            $updated_name = ["name" => $request->name];
+            $stack = Stack::where('id', $stack_id);
+
+            // check if it same user who's trying to update category
+            $user_id = $stack->first()["user_id"];
```
Comment (dasaint001) [inline backend/app/Http/Controllers/StackController.php:61]: this is error here..

File: backend/app/Http/Controllers/StackController.php [hunk 2]
```diff
@@ -12,43 +12,72 @@ class StackController extends Controller
...
+
+            if ($user_id !== $uid) {
+                return $this->sendResponse(true, "Not authorised to update this stack",
+                    "Unauthorised.", null, 404);
             }
 
-            $stack->update($updatedData);
+            if ($stack->count() == 0) {
+                return $this->sendResponse(true, "Stack doesn't exists", "Stack not found.",
+                    null, 404);
```
Comment (dasaint001) [inline backend/app/Http/Controllers/StackController.php]: use standard http response here.

File: backend/app/Http/Controllers/StackController.php [hunk 3]
```diff
@@ -12,43 +12,72 @@ class StackController extends Controller
...
-        }  catch (Exception $e) {
-            return $this->sendResponse('Stack not fetched', $e->getMessage());
+            $stack->where("user_id", $uid)->update($updated_name);
+
+            return $this->sendResponse(false, null, 'Stack updated successfully',
+                $updated_name, Response::HTTP_OK);
+        } catch (Exception $e) {
+
+            return $this->sendResponse(true, "Something went wrong updating stack " . $e->getMessage(),
+                'Failed updating stack.', null, 500);
```
Comment (dasaint001) [inline backend/app/Http/Controllers/StackController.php]: same issue here

File: backend/routes/api.php [hunk 4]
```diff
@@ -199,12 +199,12 @@ function () {
 // Stack route
 Route::prefix("stack")->group(function () {
     Route::post('add', [StackController::class, 'addStack'])->middleware("isloggedin", "isadmin");
-    Route::put('update/{stackId}', [StackController::class, 'updateStack']);
+    Route::put('update/{stackId}', [StackController::class, 'updateStack'])->middleware("isloggedin", "isadmin");
     Route::put('delete/{stackId}', [StackController::class, 'deleteStack']);
```
Comment (dasaint001) [inline backend/routes/api.php]: why put when you are deleting in this route?

File: backend/app/Http/Controllers/StackController.php [hunk 5]
```diff
@@ -12,43 +12,72 @@ class StackController extends Controller
 {
     public function addStack(StackRequest $request): JsonResponse
     {
```
Comment (dasaint001) [inline backend/app/Http/Controllers/StackController.php]: this looks like a duplicate of your formal PR.

**Review by dasaint001 — CHANGES_REQUESTED** [review]

File: backend/app/Http/Controllers/StackController.php (hunk 1 above)
Comment (Priceless-P) [reply backend/app/Http/Controllers/StackController.php:61]: What is the error?

File: backend/routes/api.php (hunk 4 above)
Comment (Priceless-P) [reply backend/routes/api.php]: That is the route for edit/update not for delete

Comment (dasaint001) [reply backend/routes/api.php]: > That is the route for edit/update not for delete

change the delete route...I commented on the delete roiute not edit route.

**Review by dasaint001 — CHANGES_REQUESTED** [review]

File: backend/routes/api.php (hunk 4 above)
Comment (Priceless-P) [reply backend/routes/api.php]: Ohh. I wasn't the one that worked on the delete route. But i will correct it like you requested

**Review by dasaint001 — CHANGES_REQUESTED** [review]
pls use the standard HTTP::response status where necessary .

**Review by dasaint001 — APPROVED** [review]

**Pull request merged**